test/extensions/KHR/SPV_KHR_untyped_pointers/store.ll

```diff
@@ -0,0 +1,49 @@
+; This test checks translation of function parameter which is untyped pointer.
+; Lately, when we do support untyped variables, this one could be used to check
+; "full" forward and reverse translation of opaque pointers.
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_KHR_untyped_pointers -o %t.spv
+; TODO: enable back once spirv-tools are updated and untyped variable is implemented.
+; R/UN: spirv-val %t.spv
+
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_KHR_untyped_pointers -spirv-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o %t.rev.ll
+; RUN: FileCheck < %t.rev.ll %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Capability UntypedPointersKHR
+; CHECK-SPIRV: Extension "SPV_KHR_untyped_pointers"
+; CHECK-SPIRV-DAG: TypeInt [[#IntTy:]] 32 0
+; CHECK-SPIRV-DAG: Constant [[#IntTy]] [[#Constant0:]] 0
+; CHECK-SPIRV-DAG: TypeUntypedPointerKHR [[#UntypedPtrTy:]] 5
+; CHECK-SPIRV-DAG: TypeFunction [[#FuncTy:]] [[#]] [[#UntypedPtrTy]]
+; CHECK-SPIRV-DAG: TypeUntypedPointerKHR [[#UntypedPtrTyFunc:]] 7
+
+; CHECK-SPIRV: Function [[#]] [[#]] 0 [[#FuncTy]]
+; CHECK-SPIRV: FunctionParameter [[#UntypedPtrTy]] [[#FuncParam:]]
+
+; CHECK-SPIRV: Variable [[#UntypedPtrTyFunc]] [[#VarBId:]] 7
+; CHECK-SPIRV: Store [[#VarBId]] [[#FuncParam]] 2 4
+; CHECK-SPIRV: Load [[#UntypedPtrTy]] [[#LoadId:]] [[#VarBId]] 2 4
+; CHECK-SPIRV: Store [[#LoadId]] [[#Constant0]] 2 4
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; CHECK-LLVM: define spir_func void @foo(ptr addrspace(1) %a)
+; CHECK-LLVM:   %b = alloca ptr, align 4
+; CHECK-LLVM:   store ptr addrspace(1) %a, ptr %b, align 4
+; CHECK-LLVM:   %0 = load ptr addrspace(1), ptr %b, align 4
+; CHECK-LLVM:   store i32 0, ptr addrspace(1) %0, align 4
+
+define spir_func void @foo(ptr addrspace(1) %a) {
+entry:
+  %b = alloca ptr addrspace(1), align 4
+  store ptr addrspace(1) %a, ptr %b, align 4
+  %0 = load ptr addrspace(1), ptr %b, align 4
+  store i32 0, ptr addrspace(1) %0, align 4
+  ret void
+}
```

Comment:
Can we also have a function `boo` in the test that also store/load non-pointer object to an opaque pointer?
like: %0 = load i32, ptr %b
I'd like to ensure, that load's type is being translated to SPIR-V correctly along with the opaque pointer type.